src/params.cpp

```diff
@@ -403,9 +403,10 @@ static int PyToCType(Cursor *cur, unsigned char **outbuf, PyObject *cell, ParamI
         pts->second = PyDateTime_DATE_GET_SECOND(cell);
 
         // Truncate the fraction according to precision
+        size_t digits = min(9, pi->DecimalDigits);
         long fast_pow10[] = {1,10,100,1000,10000,100000,1000000,10000000,100000000,1000000000};
         SQLUINTEGER milliseconds = PyDateTime_DATE_GET_MICROSECOND(cell) * 1000;
```

Comment:
This looks like a perfectly reasonable solution.  Python datetimes can only handle microseconds anyway, so any more precision than that is essentially meaningless.

Whilst we're here though, is it worth changing the name of the variable "milliseconds" to "billionths" instead?  "Milliseconds" is a confusing misnomer.  When microseconds are multiplied by 1000, the result is billionths of a second, not milliseconds.

Reply:
"nanoseconds" ref:

https://github.com/mkleehammer/pyodbc/issues/908#issuecomment-856871022

Reply:
Nanoseconds!  Yes, not "billionths".  Much better, thank you Gord.